modules/dnn/src/cuda4dnn/primitives/max_unpooling.hpp

```diff
@@ -55,8 +55,8 @@ namespace cv { namespace dnn { namespace cuda4dnn {
             strides = config.strides;
             CV_Assert(pooling_order == strides.size());
 
-            if (pooling_order != 2 && pooling_order != 3)
-                CV_Error(Error::StsNotImplemented, "Only 2D/3D max-pooling are supported.");
+            if (pooling_order > 3)
```

Comment:
Let's combine this condition with assert above:
https://github.com/opencv/opencv/blob/57809b53826db2ed2429533957ff67e994614af7/modules/dnn/src/cuda4dnn/primitives/max_unpooling.hpp#L53
```suggestion
            if (pooling_order < 1 || pooling_order > 3)
```